### PR TITLE
Remove nonsensical "child inherited" metadata

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | yes?
| New feature?  | no
| BC breaks?    | no?
| Deprecations? | yes
| Tickets       | not sure
| License       | MIT
| Doc PR        | N/A

From #3232. Targeting `master` because it's considered risky to go into a patch release.

### CODE REVIEW

**Comment by teohhanhui** [comment]
phpstan error will be fixed by https://github.com/api-platform/core/pull/3274

File: src/Serializer/AbstractItemNormalizer.php [hunk 1]
```diff
@@ -503,6 +512,7 @@ protected function getAttributeValue($object, $attribute, $format = null, array
         $context['api_attribute'] = $attribute;
         $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $attribute, $this->getFactoryOptions($context));
 
+        // BC to be removed in 3.0
```
Comment (soyuka) [inline src/Serializer/AbstractItemNormalizer.php]: what should be removed?

File: src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php [hunk 2]
```diff
@@ -48,8 +48,7 @@ public function create(string $resourceClass, string $property, array $options =
     {
         $propertyMetadata = $this->decorated->create($resourceClass, $property, $options);
 
-        // in case of a property inherited (in a child class), we need it's properties
-        // to be mapped against serialization groups instead of the parent ones.
+        // BC to be removed in 3.0
         if (null !== ($childResourceClass = $propertyMetadata->getChildInherited())) {
```
Comment (soyuka) [inline src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php]: can you specify that `childInherited` is deprecated which is why it needs to be removed?

File: src/Serializer/AbstractItemNormalizer.php (hunk 1 above)
Comment (teohhanhui) [reply src/Serializer/AbstractItemNormalizer.php]: The whole try-catch except for the getValue.

File: src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php (hunk 2 above)
Comment (teohhanhui) [reply src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php]: Hmm... We don't do this elsewhere. I think it's enough to know that this block should be removed when preparing 3. The code should not even be used under normal circumstances.

**Comment by soyuka** [comment]
can you rebase, let's merge this :)

**Comment by teohhanhui** [comment]
Rebased. :smile:

**Comment by teohhanhui** [comment]
phpstan failure seems unrelated... Looks like it's caused by the release of Symfony 5.0 :laughing:

**Pull request merged**